### ISSUE TITLE
Is it possible to send http_response asynchronously?

### ISSUE THREAD

**Issue by anlinzhang** [etr/libhttpserver#110]
All examples of render function sends http_response at the end of the function. Is it possible that the response be sent later in another user threads and let the function return for now without sending response? 
Our application requires that the response be acquired asynchronously from another entity, wondering if we can send http_response asynchronously.
There could be a workaround, such as holding the thread of render() function by waiting on a condition variable, and signal the thread to continue when the response is ready from another thread. But that requires holding the thread, and in order to server concurrent requests, we need configure the server to use one thread per http request, which we hesitate to do.

**Comment by etr** [comment]
@anlinzhang I've recently updated the documentation. Would you be able to review the example of "of a deferred response through callback" under https://github.com/etr/libhttpserver#other-examples ? Does that cover what you are trying to do?

**Comment by etr** [comment]
Closing as no feedback received for over a year.
